Bypass punning in ontodoc.

francescalb opened the pull request:
# Description:
Closes #506.
Ontodoc fails when punned individuals are included in the ontology. 
After discussions with various people (see #507)  we have agreed to not support punning for now.
This might be changed in the future if we see the need.

Tests have been added to see that ontodoc runs for ontologies with an individual, and that it runs with a warning
when the individual is a punned individual. Pytest is set to ignore that particualr warning so that this test 
will continue to pass if pytest is set to fail on warnings as well in the future.
 
## Type of change:
<!-- Put an `x` in the box that applies. -->
- [x] Bug fix.
- [ ] New feature.
- [ ] Documentation update.

## Checklist:
<!-- Put an `x` in the boxes that apply. You can also fill these out after creating the PR. -->

This checklist can be used as a help for the reviewer.

- [ ] Is the code easy to read and understand?
- [ ] Are comments for humans to read, not computers to disregard?
- [ ] Does a new feature has an accompanying new test (in the CI or unit testing schemes)?
- [ ] Has the documentation been updated as necessary?
- [ ] Does this close the issue?
- [ ] Is the change limited to the issue?
- [ ] Are errors handled for all outcomes?
- [ ] Does the new feature provide new restrictions on dependencies, and if so is this documented?

## Comments:
<!-- Additional comments here, including clarifications on checklist if applicable. -->

francescalb committed:
Added warning that punned individuals are ignored in ontodoc.py.

Also added a test to check that ontodoc actually runs with non-punned individuals and punned individuals

francescalb committed:
Added prefLabel to individual in  testonto_w_individual

francescalb committed:
Merge branch 'master' into flb/bypass_punning_in_ontodoc

jesper-friis commented on ontopy/ontodoc.py:
```diff
@@ -392,23 +392,30 @@ def itemdoc(
+                                key="Individuals", value=value, ontology=onto
+                            )
+                        )
+                except TypeError:
+                    warnings.warn(
```
Very good warning message. Are we sure that there are not other TypeErrors what will be catched by this except statement?

francescalb replied:
No, I am not sure that there are not others. But I think it is only punning that will be caught by this because we ask for an indivudal, and then we iterate of the concepts that the individual is an instance of. But I think this is a good point. If there is punning I know that is_instance_of returns 'property' and not a list of concepts, so I changed it to be a positive test for that. Better?

jesper-friis reviewed (approved):
Looks good. Added a question

francescalb committed:
Corrected test for ontodoc with punning

francescalb committed:
Merge branch 'flb/bypass_punning_in_ontodoc' of github.com:emmo-repo/EMMOntoPy into flb/bypass_punning_in_ontodoc

CasperWA committed:
Run only pytest CI (py3.7-8) on push

CasperWA committed:
Define variable prior to try/finally

CasperWA committed:
Tattle in conftest

CasperWA committed:
Fix returning tuples from os.walk()

francescalb committed:
Changed to positive test for punning, instead of TypeError

Pushed with no verify as the pylint errors are due to pylint upgrade
which finds errors in other parts of the core. This is addressed in another PR.

jesper-friis reviewed (approved):
Yes, this looks very good. 

I have not tested it locally which I normally would do if I had my linux machine at hand, but I see you have a good test.

As far as I can see, PR #535 fixes the issue with failing tests. Please merge that into this branch before merging to master.

jesper-friis reviewed (approved):
Accepted given that the test passes

CasperWA committed:
Update fixtures for testing tools

Note, only the ontodoc tests have been updated, so all other tool tests
will fail.

CasperWA committed:
Update tools tests with updated fixtures

CasperWA committed:
Revert "Run only pytest CI (py3.7-8) on push"

This reverts commit a80a96e3f44d7595b53b2628ca7c70789b48cc65.

francescalb committed:
Merge pull request #536 from emmo-repo/cwa/test_flb/bypass_punning_in_ontodoc

Fix fixtures for Python3.7.

francescalb committed:
Merge branch 'master' into flb/bypass_punning_in_ontodoc

francescalb committed:
pre-commit fix
